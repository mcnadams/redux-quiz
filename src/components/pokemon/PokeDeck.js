import React from 'react';
import PropTypes from 'prop-types';
import Pokemon from './Pokemon';

function PokeDeck({ pokemons }) {
  if(!pokemons) return <h1>Error!</h1>;

  const pokeList = pokemons.map(pokemon => {
    return (
      <li key={pokemon._id}>
        <Pokemon pokemon={pokemon} />
      </li>
    );
  });

  return (
    <ul>
      {pokeList}
    </ul>
  );
}

PokeDeck.propTypes = {
  pokemons: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    pokemon: PropTypes.string.isRequired,
    color_1: PropTypes.string.isRequired,
    color_2: PropTypes.string.isRequired,
    url_image: PropTypes.string.isRequired,
  })
};

export default PokeDeck;
