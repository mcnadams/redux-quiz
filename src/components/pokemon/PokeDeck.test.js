import { shallow } from 'enzyme';
import React from 'react';
import PokeDeck from './PokeDeck';

describe('PokeDeck component', () => {
  it('renders PokeDeck', () => {
    const wrapper = shallow(<PokeDeck pokemons={[{
      _id: '2',
      pokemon: 'bob',
      color_1: '#f0987f',
      color_2: 'NA',
      url_image: '/path',
    }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
