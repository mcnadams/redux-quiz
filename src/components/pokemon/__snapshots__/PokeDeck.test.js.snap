// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeDeck component renders PokeDeck 1`] = `
<ul
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }
>
  <li
    key="2"
    style={
      Object {
        "backgroundColor": "#f0987f",
        "border": "1px solid black",
        "borderRadius": "10px",
        "listStyleType": "none",
        "margin": "5px",
      }
    }
  >
    <Pokemon
      pokemon={
        Object {
          "_id": "2",
          "color_1": "#f0987f",
          "color_2": "NA",
          "pokemon": "bob",
          "url_image": "/path",
        }
      }
      styles={
        Object {
          "width": "80%",
        }
      }
    />
  </li>
</ul>
`;
